### DISCUSSION TITLE
Please provide form state from useForm

### DISCUSSION THREAD

**Discussion by mehrdad-shokri** [mantinedev/mantine#5201] [Q&A] [answered]
I chose to use form hooks from mantine to develop forms(I am using mantine components for ui development) and to my surprise(!) it doesn't return any info about form state. It does return isDirty and isTouched, but what about other form states? form submit(after validation passed) is usually an asynchronous operation(involving an http request mostly) and meanwhile, you want to show loading state in your UI, disable submit button(with a progress bar inside) while your form is being submitted. This is a very common and known use case for form libraries, why don't you expose such state from the hook? [This](https://react-hook-form.com/docs/useform/formstate) is form states exposed from ``react-hook-form.`` Because of this(unfortunately) I had to take a step back and install react-hook-form for my form handling which obviously is a drawback. There is also a ``isLoading`` state for use cases where form initialization is async(initial values gets loaded asynchronously). To me personally, without ``isSubmitting `` state, the whole use-form from mantine becomes useless. Thank you in advance.

**Comment by robertwt7** [comment]
yeah i need isSubmitting and isLoading as well. that has become a necessity for daily job on building forms :/

**Comment by LucasJantschChitolina** [comment]
No news?

**Comment by rtivital** [comment, marked as answer]
https://mantine.dev/form/status/#submitting-state
